apps/web/src/components/Composer/NewPublication.tsx

```diff
@@ -376,13 +416,18 @@ const NewPublication: FC<NewPublicationProps> = ({ publication }) => {
 
       // Payload for the open action module
       const openActionModules = [];
-      if (collectModule.type) {
+
+      if (Boolean(nftOpenActionEmbed)) {
+        openActionModules.push(nftOpenActionEmbed);
+      }
```

Comment:
This should be moved to the individual open action component, NewPublication.tsx should be generic

Reply:
Yeah, but this is part of the createPublication request payload.
We can add a state var and change it on the OpenAction component if there's a `nftOpenActionEmbed`
Why would we move the whole thing into it?